pkg/controller/operconfig/migration.go

```diff
@@ -97,3 +132,388 @@ func convertEgressFirewallToEgressNetworkPolicy(ctx context.Context, client cnoc
+			// - delete cloudprivateipconfig if it exists
+			// - create egressip custom resource for ovn
+			if err := deleteCloudPrivateIpConfigs(ctx, client, nns.Object["egressIPs"].([]interface{})); err != nil {
+				return nil, err
+			}
+
+			egressIpName := fmt.Sprint("egressip-", i)
+			egressIP := unstructuredEgressIpObject(egressIpName, nns.Object["egressIPs"].([]interface{}), nns.Object["netname"])
+			egressIpList = append(egressIpList, egressIP)
+		}
+	}
+
+	return egressIpList, nil // success
+}
+
+func convertOvnEgressIpToSdnEgressIp(ctx context.Context, client cnoclient.Client) error {
+	egressIpRollbackReady, err := sdnEgressIpResourcesReady(ctx, client)
+	if !egressIpRollbackReady {
+		return nil // wait for all SDN egressIP resources to be created before rollback
+	} else if err != nil {
+		return err
+	}
+
+	// 1. query for egressips
+	egressIpList, err := cnoclient.ListAllOfSpecifiedType(gvrEgressIp, ctx, client)
+	if err != nil {
+		return err
+	}
+
+	// 2. query for nodes
+	nodeList, err := client.Default().Kubernetes().CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	// 3. iterate through egressIP objects and generate netnamespaces
+	//    - value of `namespaceSelector` will determine netnamespace's name
+	//    - egressIP list will be copied to egressIPs field for netnamespace
+	for _, eip := range egressIpList {
+		// translate egressIPs to SDN netnamespaces
+		// - delete cloudprivateipconfig if it exists
+		// - create egressip to netnamespace resources for sdn
+
+		egressIps := eip.Object["spec"].(map[string]interface{})["egressIPs"].([]interface{})
+
+		if err := deleteCloudPrivateIpConfigs(ctx, client, egressIps); err != nil {
+			return err
+		}
+
+		eipNamespace, found, err := unstructured.NestedString(eip.Object, "spec", "namespaceSelector", "matchLabels", "kubernetes.io/metadata.name")
+		if !found || err != nil {
+			return fmt.Errorf("kubernetes.io/metadata.name not found in egressIP object %s, probable underlying error: %v", eip.GetName(), err)
+		}
+
+		if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+			nns, err := client.Default().Dynamic().Resource(gvrNetnamespace).Get(ctx, eipNamespace, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+
+			nns.Object["egressIPs"] = egressIps
+			_, err = client.Default().Dynamic().Resource(gvrNetnamespace).Update(ctx, nns, metav1.UpdateOptions{})
+			return err
+		}); err != nil {
+			return err
+		}
+	}
+
+	// 4. iterate through nodes and generate hostsubnets for any that are egress-assignable
+	for _, node := range nodeList.Items {
+		if node.Labels == nil {
+			continue
+		}
+		if _, ok := node.Labels[egressAssignable]; ok {
+			// generate egressCIDRs field
+			// if "egressCIDRAnnotationName" node annotation exists, automatic sdn config was used and we shall reuse old config
+			// else, manual sdn config was used and we shall use node's subnet CIDR (cannot restore egressIPs)
```

Comment:
Coming from an SDN cluster, the user will define the actual egressIP value (via hostsubnet and netnamespace). The user shouldn't be using _any node's specific IP address per se_, but should be using the node's subnet CIDR. Here's an example:

- So if some worker nodes A,B, and C have IPs `10.0.128.1`, `10.0.128.2`, and `10.0.128.3`, it is fine to define on a Netnamespace object an egressIP of `10.0.128.4`. 
- On the hostsubnet side, providing hostsubnet corresponding to worker node C with egressCIDR of `10.0.128.0/17` will allow the egressIP `10.0.128.4` to exist on worker node C. 

Providing a hostsubnet (and therefore its corresponding node) with egressCIDR equal to the node subnet CIDR effectively is the same as marking a node as "egress-assignable" in OVN. Let me know if this is clear, it can be pretty confusing.

Reply:
Thanks for explaining.